himl/inject_env.py

```diff
@@ -0,0 +1,51 @@
+# Copyright 2021 Adobe. All rights reserved.
+# This file is licensed to you under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License. You may obtain a copy
+# of the License at http://www.apache.org/licenses/LICENSE-2.0
+
+# Unless required by applicable law or agreed to in writing, software distributed under
+# the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR REPRESENTATIONS
+# OF ANY KIND, either express or implied. See the License for the specific language
+# governing permissions and limitations under the License.
+
+from os import getenv
+
+
+class EnvVarInjector(object):
+    """
+    Resolve secrets in the form:
+    {{env(HOME)}}
+    """
+
+    def __init__(self):
+        return
+
+    def is_interpolation(self, value):
+        return value.startswith('{{') and value.endswith('}}')
+
+    def inject_env_var(self, line):
+        """
+        Check if value is an interpolation and try to resolve it.
+        Uses a cache, in order to not fetch same secret multiple times.
```

Comment:
```suggestion
```

is there a cache involved?

Reply:
No there isn't, let me clean this up, elements that remained from the secrets injector that I took inspiration from.